imod_coupler/drivers/ribametamod/exchange.py

```diff
@@ -0,0 +1,200 @@
+from collections import ChainMap
+from typing import Any
+
+import numpy as np
+from numpy.typing import NDArray
+
+from imod_coupler.drivers.ribametamod.mapping import SetMapping
+from imod_coupler.kernelwrappers.mf6_wrapper import (
+    Mf6Api,
+    Mf6Drainage,
+    Mf6River,
+)
+
+
+class ExchangeBalance:
+    demands: dict[str, NDArray[np.float_]]
+    demands_negative: dict[str, NDArray[np.float_]]
+    realised_negative: dict[str, NDArray[np.float_]]
+    shape: int
+    sum_keys: list[str]
+
+    def __init__(self, shape: int, labels: list[str]) -> None:
+        self.shape = shape
+        self.flux_labels = labels
+        self._init_arrays()
+
+    def compute_realised(self, realised: NDArray[np.float_]) -> None:
+        """
+        This function computes the realised (negative) volumes
+        """
+        shortage = np.absolute(self.demand - realised)
+        demand_negative = self.demand_negative
+        self._check_valid_shortage(shortage, demand_negative)
+        # deal with zero division
+        realised_fraction = np.where(
+            demand_negative < 0.0, 1.0 - (-shortage / demand_negative), 1.0
+        )
+        for flux_label in self.flux_labels:
```

Comment:
It can be shown that we can savely do:
realised_fraction = np.where(
            self.demand_negative < 0.0, 
            (realised - self.demand + self.demand_negative) / self.demand.negative,
            1.0)
and by defiinition self.demand is less or equal realised
and self.demand equal realised if self.demand_negative is zero